nav2_controller/include/nav2_controller/nav2_controller_client.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_CONTROLLER__NAV2_CONTROLLER_CLIENT_HPP_
+#define NAV2_CONTROLLER__NAV2_CONTROLLER_CLIENT_HPP_
+
+#include <memory>
+
+#include "geometry_msgs/msg/pose_with_covariance_stamped.hpp"
+#include "geometry_msgs/msg/quaternion.hpp"
+#include "nav2_msgs/action/navigate_to_pose.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_action/rclcpp_action.hpp"
+#include "std_srvs/srv/empty.hpp"
+
+namespace nav2_controller
+{
+
+class Nav2ControllerClient
+{
+public:
+  Nav2ControllerClient();
+
+  // Client-side interface to the Nav2 controller
+  void startup();
+  void shutdown();
+  void pause();
+  void resume();
+
+  // A couple convenience methods to facilitate scripting tests
+  void set_initial_pose(double x, double y, double theta);
+  bool navigate_to_pose(double x, double y, double theta);
+
+protected:
+  using Srv = std_srvs::srv::Empty;
+
+  // A generic method used to call startup, shutdown, etc.
+  void callService(rclcpp::Client<Srv>::SharedPtr service_client, const char * service_name);
+
+  // The node to use for the service call
+  rclcpp::Node::SharedPtr node_;
+
+  // The same (empty) request for all of the services
+  std::shared_ptr<Srv::Request> request_;
```

Comment:
Suggestion: renaming `Srv` to `SrvEmpty`. This way it's easier to know that this is an empty request.

Reply:
Ok, I've used this:

`using Empty = std_srvs::srv::Empty;`

to make it clear that its the Empty service, while still providing a shortened version.